Setting debezium heartbeat.interval.ms custom property causes processing to fail [HZ-3863]

<!--
Thanks for reporting your issue. Please share with us the following information, to help us resolve your issue quickly and efficiently.
-->
  
**Describe the bug**
Setting a debezium custom property [heartbeat.interval.ms](https://debezium.io/documentation/reference/1.9/connectors/postgresql.html#postgresql-property-heartbeat-interval-ms) causes the processing to fail with the given stacktrace:

```
15:22:31,077  INFO || - [TaskletExecutionService] hz.DebeziumCdcIntegrationTest_awesome_hoover.jet.blocking.thread-0 - [127.0.0.1]:5702 [dev] [5.4.0-SNAPSHOT] Exception in ProcessorTasklet{0acf-85fb-be00-0001/postgres#0}
com.hazelcast.jet.JetException: Failed to connect to database
	at com.hazelcast.jet.cdc.impl.CdcSourceP.reconnect(CdcSourceP.java:216) ~[classes/:?]
	at com.hazelcast.jet.cdc.impl.CdcSourceP.complete(CdcSourceP.java:196) ~[classes/:?]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.complete(ProcessorTasklet.java:541) ~[classes/:?]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.stateMachineStep(ProcessorTasklet.java:421) ~[classes/:?]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.call(ProcessorTasklet.java:291) ~[classes/:?]
	at com.hazelcast.jet.impl.execution.TaskletExecutionService$BlockingWorker.run(TaskletExecutionService.java:319) [classes/:?]
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.base/java.lang.Thread.run(Thread.java:829) [?:?]
Caused by: org.apache.kafka.connect.errors.DataException: source is not a valid field name
	at org.apache.kafka.connect.data.Struct.lookupField(Struct.java:254) ~[connect-api-3.4.1.jar:?]
	at org.apache.kafka.connect.data.Struct.get(Struct.java:74) ~[connect-api-3.4.1.jar:?]
	at com.hazelcast.jet.cdc.impl.ChangeRecordCdcSourceP.map(ChangeRecordCdcSourceP.java:66) ~[classes/:?]
	at com.hazelcast.jet.cdc.impl.ChangeRecordCdcSourceP.map(ChangeRecordCdcSourceP.java:34) ~[classes/:?]
	at com.hazelcast.jet.cdc.impl.CdcSourceP.lambda$complete$0(CdcSourceP.java:187) ~[classes/:?]
	at com.hazelcast.jet.impl.util.FlatMappingTraverser.nextTraverser(FlatMappingTraverser.java:63) ~[classes/:?]
	at com.hazelcast.jet.impl.util.FlatMappingTraverser.next(FlatMappingTraverser.java:55) ~[classes/:?]
	at com.hazelcast.jet.core.AbstractProcessor.emitFromTraverser(AbstractProcessor.java:407) ~[classes/:?]
	at com.hazelcast.jet.core.AbstractProcessor.emitFromTraverser(AbstractProcessor.java:421) ~[classes/:?]
	at com.hazelcast.jet.cdc.impl.CdcSourceP.complete(CdcSourceP.java:191) ~[classes/:?]
	... 9 more
```

**Expected behavior**
Hazelcast should somehow be able to process heartbeat events from debezium once they are set up.

**To Reproduce**

Extend some of the integration tests, e.g. `com.hazelcast.jet.cdc.DebeziumCdcIntegrationTest#postgres` with `heartbeat.interval.ms` property set to some non-null value

```java
[...]
                    .setProperty("table.whitelist", "inventory.customers")
                    .setProperty("heartbeat.interval.ms", "1000") // <- set the property to e.g. 1sec
                    .build();
[...]
```
Additionally, add to hz instance config:
```java
config.setProperty("hazelcast.logging.type", "log4j2");
```

Now once the test is run we should be able to see in the logs the `DataException` error.

--- a/extensions/cdc-debezium/src/main/java/com/hazelcast/jet/cdc/impl/ChangeRecordCdcSourceP.java
+++ b/extensions/cdc-debezium/src/main/java/com/hazelcast/jet/cdc/impl/ChangeRecordCdcSourceP.java
@@ -54,7 +54,8 @@ public class ChangeRecordCdcSourceP extends CdcSourceP<ChangeRecord> {
     @Nullable
     @Override
     protected ChangeRecord map(SourceRecord record) {
-        if (record == null || record.topic().startsWith("__debezium")) { // internal Debezium messages
+        if (record == null || record.topic().startsWith("__debezium")) {
+            // internal Debezium messages about e.g. Heartbeat uses such topics
             return null;
         }
 
@@ -65,7 +66,9 @@ public class ChangeRecordCdcSourceP extends CdcSourceP<ChangeRecord> {
         Schema valueSchema = record.valueSchema();
 
         if (valueSchema.name().startsWith("io.debezium.")) {
-            return null; // internal Debezium messages
+            // internal Debezium messages, e.g. transaction metadata uses topic ${database.server.name}.transaction
+            // so it won't be filtered out earlier
+            return null;
         }
         Struct source = (Struct) value.get("source");
 

--- a/extensions/cdc-debezium/src/main/java/com/hazelcast/jet/cdc/impl/ChangeRecordCdcSourceP.java
+++ b/extensions/cdc-debezium/src/main/java/com/hazelcast/jet/cdc/impl/ChangeRecordCdcSourceP.java
@@ -54,7 +54,7 @@ public class ChangeRecordCdcSourceP extends CdcSourceP<ChangeRecord> {
     @Nullable
     @Override
     protected ChangeRecord map(SourceRecord record) {
-        if (record == null) {
+        if (record == null || record.topic().startsWith("__debezium")) { // internal Debezium messages
             return null;
         }
 

--- a/extensions/cdc-debezium/src/test/java/com/hazelcast/jet/cdc/DebeziumCdcIntegrationTest.java
+++ b/extensions/cdc-debezium/src/test/java/com/hazelcast/jet/cdc/DebeziumCdcIntegrationTest.java
@@ -264,6 +264,7 @@ public class DebeziumCdcIntegrationTest extends AbstractCdcIntegrationTest {
                     .setProperty("database.password", "postgres")
                     .setProperty("database.dbname", "postgres")
                     .setProperty("table.whitelist", "inventory.customers")
+                    .setProperty("heartbeat.interval.ms", "1000") // this will add Heartbeat messages to the stream
                     .build();
 
             Pipeline pipeline = Pipeline.create();

